### PR TITLE
Add DiffService Changes to OS4

### PR DESCRIPTION
**Pull request by tsiegleauq**
Integrate more recent Diff Service Changes to OS4.
Not totally compatible with #101, some adjustmens have
been made

Includes:
https://github.com/OpenSlides/OpenSlides/commit/ba0e9b3bc6b1f96f66a98cc5b20ab1739d60db81
https://github.com/OpenSlides/OpenSlides/commit/ac50d6f8dccc5b502c7b7a3a106ef5935184494f
https://github.com/OpenSlides/OpenSlides/commit/ba177a89d49da59f9cfe56b96b82c7a2f4040b7e
https://github.com/OpenSlides/OpenSlides/commit/7275aa69af3fdbd14b407e2b6200eacb5aa9b39a
https://github.com/OpenSlides/OpenSlides/pull/6047/files

### CODE REVIEW

File: client/src/app/site/motions/modules/motion-detail/components/motion-content/motion-content.component.ts [hunk 1]
```diff
@@ -429,6 +435,20 @@ export class MotionContentComponent extends BaseComponent implements OnInit, OnD
...
+             */
+            // for (
+            //     let paragraphNo = paragraphsToChoose.length;
+            //     paragraphNo < formMotion.amendment_paragraphs.length;
+            //     paragraphNo++
+            // ) {
+            //     if (formMotion.amendment_paragraphs[paragraphNo] !== null) {
+            //         contentPatch.broken_paragraphs.push(formMotion.amendment_paragraphs[paragraphNo]);
+            //     }
+            // }
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/motion-detail/components/motion-content/motion-content.component.ts:451]: @GabrielInTheWorld relevant for checking, not obvious how OS4 motions encapsulate lists of motion amendments

File: client/src/app/core/repositories/motions/motion-line-numbering.service.ts [hunk 2]
```diff
@@ -485,6 +503,10 @@ export class MotionLineNumberingService {
 
         return baseParagraphs.map((origText: string, paraNo: number): string => {
             const newText = amendment.amendment_paragraph(paraNo);
+            /**
+             * TODO CHECK please check if this commit was coverd justly
+             * https://github.com/OpenSlides/OpenSlides/commit/ac50d6f8dccc5b502c7b7a3a106ef5935184494f#diff-f5bbc561818c722671cf3105c1c578d9e6b9767bb0f4f9ea32030678eb803499R1016-R1018
+             */
```
Comment (tsiegleauq) [inline client/src/app/core/repositories/motions/motion-line-numbering.service.ts:509]: This part seems not compatible with #101 
@GabrielInTheWorld please check
https://github.com/OpenSlides/OpenSlides/commit/ac50d6f8dccc5b502c7b7a3a106ef5935184494f#diff-f5bbc561818c722671cf3105c1c578d9e6b9767bb0f4f9ea32030678eb803499R1016-R1018

and make according changes

**Comment by GabrielInTheWorld** [comment]
@tsiegleauq I have made changes and pushed to your branch. Please verify and give me a feedback, if something is missing.

**Comment by tsiegleauq** [comment]
@GabrielInTheWorld the changes were fine. Apparently you were refactoring functions to a point where they don't fit with tobias' changes anymore? In this case, please take over this pr

**Pull request merged**